lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -2240,21 +2254,52 @@ deletePendingOrExpiredTx wid tid = do
         Nothing -> fromIntegral <$> deleteWhereCount
             ((TxMetaStatus <-. [W.Pending, W.Expired]):filt)
 
+listPendingLocalTxSubmissionQuery
+    :: W.WalletId
+    -> SqlPersistT IO [(W.SlotNo, LocalTxSubmission)]
+listPendingLocalTxSubmissionQuery wid = fmap unRaw <$> rawSql query params
+  where
+    query = "SELECT meta.slot,?? FROM (select tx_id, status, slot, MAX(slot) AS latest_slot FROM tx_meta WHERE wallet_id=? GROUP by tx_id) AS meta INNER JOIN local_tx_submission ON meta.tx_id=local_tx_submission.tx_id WHERE status=? AND slot=latest_slot"
```

Comment:
This `MAX(slot)` suggests that we are now storing multiple metadata per transaction? This raises an alarm in my head for some reasons I am not yet able to explain. I'd like to review the implication of this a bit more carefully (will do in a next review).

Reply:
Oops, this check isn't needed. There is only ever one TxMeta per txid per wallet. In the case of rollback, the status field is mutated.